### PR TITLE
Add python dependabot using pip-compile

### LINKED ISSUE #1697: Pin backend requirements and set up dependabot

### PR DESCRIPTION
**Pull request by aapeliv**
Closes #1697. Thanks to @shuttle1987 for showing me how to do this elsewhere!

### CODE REVIEW

**Comment by aapeliv** [comment]
@kalvdans I can also add you as a reviewer if you'd like

File: .github/dependabot.yml [hunk 1]
```diff
@@ -10,3 +11,23 @@ updates:
     reviewers:
       - "darrenvong"
       - "lucaslcode"
+
+  - package-ecosystem: "pip-compile"
```
Comment (kalvdans) [inline .github/dependabot.yml]: According to https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates#package-ecosystem the value should be `"pip"`, not `"pip-compile"`.

I'm all for pinned versions, great!

Comment (aapeliv) [reply .github/dependabot.yml]: Thanks, not sure how I screwed that up!

**Comment by kalvdans** [comment]
How does github know which branch(es) to monitor? i.e. `develop` instead of `master`

**Comment by aapeliv** [comment]
> How does github know which branch(es) to monitor? i.e. `develop` instead of `master`

It uses the default branch, apparently: https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates#target-branch

**Review by kalvdans — APPROVED** [review]

**Pull request merged**